[Merged by Bors] - require http and metrics for respective flags

jxs opened the pull request:
## Issue Addressed

following discussion on https://github.com/sigp/lighthouse/pull/4639#discussion_r1305183750 this PR makes the `http` and `metrics` sub-flags to require those main flags enabled

jxs committed:
require http and metrics for respective flags

jxs committed:
update default to default_if to comply with requires

jxs committed:
remove unrequired required flags

default is false

jimmygchen reviewed (approved):
LGTM!

jimmygchen reviewed (changes requested):
Oh sorry, just realized we should probably make the same changes to `beacon_node/src/cli.rs`?

jxs committed:
require http flag for http subflags for beacon node

jxs commented:
> Oh sorry, just realized we should probably make the same changes to `beacon_node/src/cli.rs`?

yup makes sense Jimmy, updated!

jimmygchen commented on beacon_node/src/cli.rs:
```diff
@@ -316,22 +316,25 @@ pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
         .arg(
             Arg::with_name("http-address")
                 .long("http-address")
+                .requires("http")
```
Hey sorry @jxs, I've just realized that http api can be enabled via `--staking` or `--gui` as well, so `--http` is not strictly necessary, so this may be a breaking change for the beacon node, and we might want to leave this unchnaged?

https://github.com/sigp/lighthouse/blob/d61f5071849723f56c4cdd02e8a4f0c6100d1543/beacon_node/src/config.rs#L88-L91

https://github.com/sigp/lighthouse/blob/d61f5071849723f56c4cdd02e8a4f0c6100d1543/beacon_node/src/config.rs#L810-L813

jimmygchen replied:
btw I discovered this from the failing `doppelganger` tests, it uses `--staking` instead of `--http`

jxs replied:
thanks for digging it Jimmy! I updated with a `ArgGroup` for `http`, `gui` and `staking`, when one of them is missing the following error is displayed:
```
error: The following required arguments were not provided:
    <--http|--gui|--staking>
```
Do you think it's clear enough? If not I can revert, meanwhile the `doppelganger` tests are still failing, can you access the logs? I am able to run the `beacon_node` for them locally but get a jwt validation error instead

jxs replied:
ok managed to fix the `doppelganger protection tests` and it seems to me it's mo longer a breaking change since all the flags still work as only one of the three is required, but ptal Jimmy

jxs committed:
make enable_http a group with http, gui, and staking

michaelsproul commented:
Marking this as `backwards-incompat`. We shouldn't merge this for v4.4.2, and should wait for v4.5.0.

jxs committed:
check for enable_http arggroup instead of http,

to both add default values and then require them, fixes doppelganger protection test

michaelsproul reviewed (approved):
Nice! Let's merge

paulhauner commented:
bors r+

michaelsproul commented:
bors r-
bors r+
